pages/api/heroes/[heroId].js

```diff
@@ -1,31 +1,60 @@
 import { PrismaClient } from '@prisma/client';
-import { authenticated } from '../../../utils/authentication';
+import { authenticated } from '../../../utils';
 
 const prisma = new PrismaClient();
 
 export default authenticated(async function handler(req, res) {
-  const { heroId } = req.query;
+  let { heroId } = req.query;
+  heroId = parseInt(heroId);
+
   if (req.method === 'GET') {
     // GET METHOD
-    const heroData = await prisma.heroes.findUnique({
-      where: {
-        id: parseInt(heroId),
-      },
-    });
-    res.status(200).json(heroData);
+    try {
+      const result = await prisma.heroes.findUnique({
+        where: {
+          id: heroId,
+        },
+        include: {
+          skills: true,
+        },
+      });
+      res.status(200).json(result);
+    } catch (error) {
+      res.status(500).json({ message: 'Cannot get any data' });
+    }
   } else if (req.method === 'PATCH') {
     // PATCH METHOD
-    const updatedData = req.body;
+    const { name, type, skills } = req.body;
+    const skillSet = [];
+    if (skills) {
+      skills.forEach((skill) => {
+        skillSet.push({ title: skill.toLowerCase() });
+      });
+    }
     try {
-      const heroes = await prisma.heroes.update({
+      const result = await prisma.heroes.update({
         where: {
-          id: parseInt(heroId),
+          id: heroId,
+        },
+        data: {
+          name,
+          type,
+          skills: {
+            connectOrCreate: skillSet.map((skill) => {
+              return {
+                where: skill.toLowerCase(),
```

Comment:
```suggestion
                where: skill.title.toLowerCase(),
```

ini bukannya harus di akses dulu property nya mas @berry1611 ?
karena kan di atas itu dia dimasukkin ke array of objects gitu